### PR TITLE
fix(op_crates/web): Use WorkerLocation for location in workers

### PR DESCRIPTION
**Pull request by nayeemrmn**
- fix(op_crates/web): Strip username and password from location

---

Includes #9044.

### CODE REVIEW

File: cli/tests/070_location.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
-// TODO(nayeemrmn): Add `Location` and `location` to `dlint`'s globals.
+// TODO(nayeemrmn): Add `Location`, `WorkerLocation` and `location` to `dlint`'s
+// globals.
```
Comment (lucacasonato) [inline cli/tests/070_location.ts:2]: Done in https://github.com/denoland/deno_lint/pull/586

File: cli/tests/071_location_unset.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-// TODO(nayeemrmn): Add `Location` and `location` to `dlint`'s globals.
+// TODO(nayeemrmn): Add `Location`, `WorkerLocation` and `location` to `dlint`'s
+// globals.
```
Comment (lucacasonato) [inline cli/tests/071_location_unset.ts:2]: ditto

File: cli/tests/079_location_authentication.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// TODO(nayeemrmn): Add `Location`, `WorkerLocation` and `location` to `dlint`'s
+// globals.
```
Comment (lucacasonato) [inline cli/tests/079_location_authentication.ts:2]: ditto

File: cli/tests/subdir/worker_location.ts [hunk 4]
```diff
@@ -1,4 +1,9 @@
+// TODO(nayeemrmn): Add `Location`, `WorkerLocation` and `location` to `dlint`'s
+// globals.
+// deno-lint-ignore-file no-undef
```
Comment (lucacasonato) [inline cli/tests/subdir/worker_location.ts:3]: ditto

File: cli/tests/workers_test.ts [hunk 5]
```diff
@@ -660,7 +660,8 @@ Deno.test({
 Deno.test({
   name: "worker with relative specifier",
   fn: async function (): Promise<void> {
-    // TODO(nayeemrmn): Add `Location` and `location` to `dlint`'s globals.
+    // TODO(nayeemrmn): Add `Location`, `WorkerLocation` and `location` to `dlint`'s
+    // globals.
```
Comment (lucacasonato) [inline cli/tests/workers_test.ts:664]: ditto

**Review by lucacasonato — COMMENTED** [review]
Looks good, just some minor comments.

File: cli/tests/070_location.ts (hunk 1 above)
Comment (nayeemrmn) [reply cli/tests/070_location.ts:2]: @lucacasonato So is this PR blocked by a dlint release and `third_party` update?

Comment (bartlomieju) [reply cli/tests/070_location.ts:2]: No, but we should release new version of `deno_lint` crate before `1.7.0`, I'll look into that

Comment (bartlomieju) [reply cli/tests/070_location.ts:2]: Let's wait on https://github.com/denoland/deno/pull/9127 then the TODOs can be removed

Comment (nayeemrmn) [reply cli/tests/070_location.ts:2]: We also need to update `dlint` in `third_party`.

Comment (bartlomieju) [reply cli/tests/070_location.ts:2]: Got you covered in https://github.com/denoland/deno/pull/9134

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**